backend/jottacloud/jottacloud.go

```diff
@@ -228,6 +239,47 @@ func shouldRetry(resp *http.Response, err error) (bool, error) {
 	return fserrors.ShouldRetry(err) || fserrors.ShouldRetryHTTP(resp, retryErrorCodes), err
 }
 
+func teliaCloudConfig(ctx context.Context, name string, m configmap.Mapper) {
+	teliaCloudOauthConfig := &oauth2.Config{
+		Endpoint: oauth2.Endpoint{
+			AuthURL:  teliaCloudAuthURL,
+			TokenURL: teliaCloudTokenURL,
+		},
+		ClientID:    teliaCloudClientID,
+		Scopes:      []string{"openid", "jotta-default", "offline_access"},
+		RedirectURL: oauthutil.RedirectLocalhostURL,
+	}
+
+	err := oauthutil.Config(ctx, "jottacloud", name, m, teliaCloudOauthConfig, nil)
+	if err != nil {
+		log.Fatalf("Failed to configure token: %v", err)
+		return
+	}
+
+	fmt.Printf("\nDo you want to use a non standard device/mountpoint e.g. for accessing files uploaded using the official Jottacloud client?\n\n")
+	if config.Confirm(false) {
+		oAuthClient, _, err := oauthutil.NewClient(ctx, name, m, teliaCloudOauthConfig)
+		if err != nil {
+			log.Fatalf("Failed to load oAuthClient: %s", err)
+		}
+
+		srv := rest.NewClient(oAuthClient).SetRoot(rootURL)
+		apiSrv := rest.NewClient(oAuthClient).SetRoot(apiURL)
+
+		device, mountpoint, err := setupMountpoint(ctx, srv, apiSrv)
+		if err != nil {
+			log.Fatalf("Failed to setup mountpoint: %s", err)
+		}
+		m.Set(configDevice, device)
+		m.Set(configMountpoint, mountpoint)
+	}
+
+	m.Set("configVersion", strconv.Itoa(configVersion))
+	m.Set(configClientID, teliaCloudClientID)
+	m.Set(configClientSecret, "")
+	m.Set(configTokenURL, teliaCloudTokenURL)
```

Comment:
You are setting clientID and clientSecret here but using configVersion 1 (which is v2 config) which means those aren't actually loaded. Does refresh actually work?

The config Version stuff is turning out kinda messy I might have to rethink that a bit at some point.

Reply:
Yeah it threw me off at first that v1configVersion is actually 0 and configVersion is 1 :)

Refresh works with non-legacy (i.e. configVersion = 1). I confirmed it works when the client_id is specified in the config but doesn't work if it's missing. I didn't go through the whole flow to verify if this is just due to a lucky side effect of unintentional behaviour though. FWIW it's worked for me without problems for a month.

Reply:
Thinking about it clientID and clientSecret aren't even part of the normal OAuth refresh flow. I know it's just a nitpick but setting them here gives the suggestion of them being used when in reality they aren't (unless I'm grossly mistaken or missed an obvious bug) so I'd suggest removing them here and checking if it still works.

Besides this minor nitpick this looks good.

I'm kinda

Reply:
I can confirm that it doesn't work without `client_id` but works with it. It's trying to send the wrong token as refresh token when no `client_id` is specified. Will investigate further why it does that and circle back.

Reply:
Ok, so correcting myself, the token is the same but the `client_id` is passed in the refresh request, first in an `Authorization: Basic` header, and if that fails it tries to issue another request where `client_id` is specified in the POST data instead. So, when `client_id` isn't set the default value `jottacli` will be used. This client is not valid for the `telia_se` realm used by Telia Cloud.

You're right that `client_secret` isn't needed though, so it can be omitted. Will push a commit for that.

Reply:
Oh yeah I forgot about the auth header. Sorry

Reply:
Oh yeah sorry forgot about the Auth header.